### PR TITLE
Specify patchable dates

### PR DESCRIPTION
**Pull request by NobodysNightmare**
## Work Packages
- https://community.openproject.org/work_packages/17417
- https://community.openproject.org/work_packages/17418
## Changes
- clearer wording regarding time related types
  - start date and due date are now **date only**, no time component
  - creation and change dates are **date and time**
  - estimated and spent time are **durations**
- Project ressource now uses `createdAt` and `updatedAt` instead of `createdOn` and `updatedOn`
- Dates are now patchable

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://feature-specify-patchable-dates.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: doc/apiv3-documentation.apib [hunk 1]
```diff
@@ -1657,10 +1657,7 @@ The request body is the actual string that shall be rendered as HTML string.
                 "priority": "Normal",
                 "startDate": null,
                 "dueDate": null,
-                "estimatedTime": {
-                    "units": "hours",
-                    "value": null
-                },
+                "estimatedTime": PT2H,
```
Comment (myabc) [inline doc/apiv3-documentation.apib]: @NobodysNightmare missing `"`

File: doc/apiv3-documentation.apib [hunk 2]
```diff
@@ -1536,13 +1536,13 @@ The request body is the actual string that shall be rendered as HTML string.
...
 | parentId | Parent work package id | Integer | Must be an id of an existing and visible (in respect to the API user) work package | 42 | READ / WRITE |
 | priority | | String | Must be one of the activated priorities | High | READ |
-| startDate | | Date | Must be a date in format YYYY-MM-DD and must be equal or greater than the soonest possible start date | 2014-05-21T08:51:20Z | READ |
-| dueDate | | Date | Must be a date in format YYYY-MM-DD and must be greater then start date | | READ |
-| estimatedTime | | Object | ISO 8601 duration | P1DT18H  | READ |
-| spentTime | | Object | ISO 8601 duration | PT1H | READ |
+| startDate | | Date | ISO 8601 date; must be equal or greater than the soonest possible start date | 2014-05-21 | READ / WRITE |
+| dueDate | | Date | ISO 8601 date; must be greater than start date | 2014-05-22 | READ / WRITE |
+| estimatedTime | | Duration | ISO 8601 duration | P1DT18H  | READ |
+| spentTime | | Duration | ISO 8601 duration | PT1H | READ |
```
Comment (myabc) [inline doc/apiv3-documentation.apib]: @ulferts @NobodysNightmare How should we handle attributes that are added by common plugins (e.g. `spentTime` from costs)? I think we should document them - probably here too, but differentiated somehow.

Comment (ulferts) [inline doc/apiv3-documentation.apib]: I would rather document them in the plugins. In the worst case, we have two plugins adding the same attribute name but those having different meanings, e.g values. Additionally, I wouldn't want OpenProject to know to much about the plugins.

FWIW, `spentTime` is an attribute of OpenProject itself. Costs adds `spentHours` (for now: https://github.com/finnlabs/openproject-costs/pull/114)  and others.

**Comment by NobodysNightmare** [comment]
tables are becoming too wide...

WP properties are not fully visible anymore, columns are pretty narrow... and one could argue that 260 chars is too much ^^

edit: Probably not a problem of this PR... but there will be the need to do something about the tables...

**Pull request merged**